src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -580,22 +578,12 @@ private Set<ProgrammingExerciseTestCase> filterTestCasesForStudents(final Set<Pr
      */
     private Result calculateScoreForResult(Set<ProgrammingExerciseTestCase> testCases, Set<ProgrammingExerciseTestCase> testCasesForCurrentDate, @NotNull Result result,
             ProgrammingExercise exercise, boolean applySubmissionPolicy) {
-        List<Feedback> testCaseFeedback = new ArrayList<>();
-        List<Feedback> staticCodeAnalysisFeedback = new ArrayList<>();
-        for (var feedback : result.getFeedbacks()) {
-            if (feedback.getType() != FeedbackType.AUTOMATIC) {
-                continue;
-            }
+        Map<Boolean, List<Feedback>> feedbacksGroupedByIsSCA = result.getFeedbacks().stream().filter(feedback -> FeedbackType.AUTOMATIC.equals(feedback.getType()))
+                .collect(Collectors.groupingBy(Feedback::isStaticCodeAnalysisFeedback));
+        List<Feedback> staticCodeAnalysisFeedback = feedbacksGroupedByIsSCA.getOrDefault(true, new ArrayList<>());
+        List<Feedback> testCaseFeedback = feedbacksGroupedByIsSCA.getOrDefault(false, new ArrayList<>()); // Feedbacks that are not SCA here are test cases
```

Comment:
The old approach seamed more intuitive to me. Creating a map just to delete it right afterwards seams a bit unnecessary